api/v1alpha1/profilecatalogsource_types.go

```diff
@@ -19,13 +19,30 @@ package v1alpha1
 import (
 	"strings"
 
+	"github.com/fluxcd/pkg/apis/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ProfileCatalogSourceSpec defines the desired state of ProfileCatalogSource
 type ProfileCatalogSourceSpec struct {
 	// Profiles is the list of profiles exposed by the catalog
+	// +optional
 	Profiles []ProfileCatalogEntry `json:"profiles,omitempty"`
+	// Repositories is the list of repositories to scan for profiles
+	// +optional
+	Repos []Repository `json:"repositories,omitempty"`
+}
+
+type Repository struct {
+	// URL is the URL of the repository
```

Comment:
Can we expand on this? Like, what format, can it be private, public, ssh or https...